src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -35,24 +35,32 @@ def publish_offers_bids(self, message):
         published data are of the following format
         market_offers_bids_list_mapping = {"market_id" : {"bids": [], "offers": [] }, }
         """
-        # TODO: message can contain filters
-        data = {"event": "offers_bids_response"}
+        response_data = {"event": "offers_bids_response"}
+        data = json.loads(message.get("data"))
+        filters = data.get("filters", {})
+        # IDs of markets the client is interested in
+        market_ids_in_interest = filters.get("markets", None)
         market_offers_bids_list_mapping = {}
         for area_uuid, markets in self.area_uuid_markets_mapping.items():
+            if market_ids_in_interest and area_uuid not in market_ids_in_interest:
+                # Client is uninterested in the market -> skip
+                continue
             for market in markets:
+                # Cache the market (needed while matching)
                 self.markets_mapping[market.id] = market
+
                 market_offers_bids_list_mapping[market.id] = {"bids": [], "offers": []}
                 bids, offers = market.open_bids_and_offers
                 market_offers_bids_list_mapping[market.id]["bids"].extend(
                     list(bid.serializable_dict() for bid in bids.values()))
                 market_offers_bids_list_mapping[market.id]["offers"].extend(
                     list(offer.serializable_dict() for offer in offers.values()))
-        data.update({
-            "market_offers_bids_list_mapping": market_offers_bids_list_mapping,
+        response_data.update({
+            "orders": market_offers_bids_list_mapping,
```

Comment:
I agree, that `market_offers_bids_list_mapping` could be rephrased, but `orders` is not very descriptive IMO. Is this a requirement or can we change it to something like: `offer_bid_lists_in_markets` ( I know, not better, but maybe somebody else has a better idea)?

Reply:
I agree, I'm open to suggestions here

Reply:
I dont have any and vote to keep `market_offers_bids_list_mapping ` as you changed it back.